CALCULATOR_UPDATED.c

```diff
@@ -0,0 +1,164 @@
+#include<stdio.h>
+#include<stdlib.h> 
+#include<stdbool.h>
+
+#define OPERATORS "*/+-"  
+
+void readInput(char user_input[]) 
+{
+    printf("Enter a expression to be calculated: ");
+    fgets(user_input,100,stdin);
+    //If i press enter after typing input,it adds \n in string and gives error
+    // here need to replace \n by null  character
+     for (int i = 0; user_input[i] != '\0'; i++)

```

Comment:
can we use iterator instead of i? and also define macro null character as it is used in many places.